File: Desktop/GESTION-MAGASIN-projet-c-cpp-python-java/projet4.php
<?php

define("FICHIER", "stock.txt");

class Produit {
    public $nom;
    public $categorie;
    public $prix;
    public $quantite;

    public function __construct($nom, $categorie, $prix, $quantite) {
        $this->nom = $nom;
        $this->categorie = $categorie;
        $this->prix = $prix;
        $this->quantite = $quantite;
    }

    public function __toString() {
        return "Nom: $this->nom, Catégorie: $this->categorie, Prix: $this->prix, Quantité: $this->quantite";
    }
}

// Liste pour stocker les produits
$stock = [];

// Charger les produits depuis un fichier
function chargerStock() {
    global $stock;
    if (!file_exists(FICHIER)) {
        echo "Aucun fichier trouvé. Un nouveau fichier sera créé.\n";
        return;
    }

    $lines = file(FICHIER, FILE_IGNORE_NEW_LINES);
    foreach ($lines as $line) {
        list($nom, $categorie, $prix, $quantite) = explode(",", $line);
        $stock[] = new Produit($nom, $categorie, (float)$prix, (int)$quantite);
    }
}

// Sauvegarder les produits dans un fichier
function sauvegarderStock() {
    global $stock;
    $lines = [];
    foreach ($stock as $produit) {
        $lines[] = "$produit->nom,$produit->categorie,$produit->prix,$produit->quantite";
    }
    file_put_contents(FICHIER, implode(PHP_EOL, $lines));
    echo "Stock sauvegardé avec succès.\n";
}

// Ajouter un produit
function ajouterProduit() {
    global $stock;
    echo "Entrez le nom : ";
    $nom = trim(fgets(STDIN));
    echo "Entrez la catégorie : ";
    $categorie = trim(fgets(STDIN));
    echo "Entrez le prix : ";
    $prix = (float)trim(fgets(STDIN));
    echo "Entrez la quantité : ";
    $quantite = (int)trim(fgets(STDIN));

    $stock[] = new Produit($nom, $categorie, $prix, $quantite);
    echo "Produit ajouté avec succès !\n";
}

// Afficher le stock
function afficherStock() {
    global $stock;
    if (empty($stock)) {
        echo "Le stock est vide.\n";
        return;
    }

    echo "\n--- État actuel du stock ---\n";
    foreach ($stock as $index => $produit) {
        echo ($index + 1) . ". $produit\n";
    }
}

// Rechercher un produit par nom
function rechercherProduit() {
    global $stock;
    echo "Entrez le nom du produit à rechercher : ";
    $nom = trim(fgets(STDIN));

    foreach ($stock as $produit) {
        if (strcasecmp($produit->nom, $nom) === 0) {
            echo "Produit trouvé : $produit\n";
            return;
        }
    }
    echo "Produit non trouvé.\n";
}

// Modifier un produit
function modifierProduit() {
    global $stock;
    echo "Entrez le nom du produit à modifier : ";
    $nom = trim(fgets(STDIN));

    foreach ($stock as $produit) {
        if (strcasecmp($produit->nom, $nom) === 0) {
            echo "Entrez le nouveau nom : ";
            $produit->nom = trim(fgets(STDIN));
            echo "Entrez la nouvelle catégorie : ";
            $produit->categorie = trim(fgets(STDIN));
            echo "Entrez le nouveau prix : ";
            $produit->prix = (float)trim(fgets(STDIN));
            echo "Entrez la nouvelle quantité : ";
            $produit->quantite = (int)trim(fgets(STDIN));
            echo "Produit modifié avec succès.\n";
            return;
        }
    }
    echo "Produit non trouvé.\n";
}

// Supprimer un produit
function supprimerProduit() {
    global $stock;
    echo "Entrez le nom du produit à supprimer : ";
    $nom = trim(fgets(STDIN));

    foreach ($stock as $index => $produit) {
        if (strcasecmp($produit->nom, $nom) === 0) {
            array_splice($stock, $index, 1);
            echo "Produit supprimé avec succès.\n";
            return;
        }
    }
    echo "Produit non trouvé.\n";
}

// Mettre à jour la quantité d'un produit
function mettreAJourQuantite() {
    global $stock;
    echo "Entrez le nom du produit : ";
    $nom = trim(fgets(STDIN));

    foreach ($stock as $produit) {
        if (strcasecmp($produit->nom, $nom) === 0) {
            echo "Entrez la nouvelle quantité : ";
            $produit->quantite = (int)trim(fgets(STDIN));
            echo "Quantité mise à jour avec succès.\n";
            return;
        }
    }
    echo "Produit non trouvé.\n";
}

// Afficher le menu principal
function afficherMenu() {
    echo "\n******************************************\n";
    echo "*         Gestion de Stock Magasin       *\n";
    echo "******************************************\n";
    echo "* 1. Ajouter un produit                  *\n";
    echo "* 2. Afficher le stock                   *\n";
    echo "* 3. Sauvegarder le stock                *\n";
    echo "* 4. Modifier un produit                 *\n";
    echo "* 5. Supprimer un produit                *\n";
    echo "* 6. Rechercher un produit               *\n";
    echo "* 7. Mettre à jour la quantité           *\n";
    echo "* 8. Quitter                             *\n";
    echo "******************************************\n";
}

// Programme principal
function main() {
    chargerStock();
    do {
        afficherMenu();
        echo "Entrez votre choix : ";
        $choix = trim(fgets(STDIN));

        switch ($choix) {
            case "1":
                ajouterProduit();
                break;
            case "2":
                afficherStock();
                break;
            case "3":
                sauvegarderStock();
                break;
            case "4":
                modifierProduit();
                break;
            case "5":
                supprimerProduit();
                break;
            case "6":
                rechercherProduit();
                break;
            case "7":
                mettreAJourQuantite();
                break;
            case "8":
                echo "Merci d'avoir utilisé le programme. Au revoir !\n";
                sauvegarderStock();
                exit;
            default:
                echo "Choix invalide, veuillez réessayer.\n";
        }
    } while (true);
}

// Exécution du programme
main();

File: Desktop/GESTION-MAGASIN-projet-c-cpp-python-java/projet4.py
import os

# Structure des données pour un produit
class Produit:
    def __init__(self, nom, categorie, prix, quantite):
        self.nom = nom
        self.categorie = categorie
        self.prix = prix
        self.quantite = quantite

    def __str__(self):
        return f"Nom: {self.nom}, Catégorie: {self.categorie}, Prix: {self.prix:.2f}, Quantité: {self.quantite}"


# Liste pour stocker les produits
stock = []

# Fichier pour la persistance
FICHIER = "stock.txt"


# Fonction pour afficher le menu
def afficher_menu():
    print("\n******************************************")
    print("*         Gestion de Stock Magasin       *")
    print("******************************************")
    print("* 1. Ajouter un produit                  *")
    print("* 2. Afficher le stock                   *")
    print("* 3. Sauvegarder le stock                *")
    print("* 4. Modifier un produit                 *")
    print("* 5. Supprimer un produit                *")
    print("* 6. Rechercher un produit               *")
    print("* 7. Mettre à jour la quantité           *")
    print("* 8. Quitter                             *")
    print("******************************************")


# Charger les données depuis un fichier
def charger_stock():
    if not os.path.exists(FICHIER):
        print(f"Fichier '{FICHIER}' introuvable. Un nouveau fichier sera créé.")
        return

    with open(FICHIER, "r") as f:
        for ligne in f:
            parts = ligne.strip().split(",")
            if len(parts) == 4:
                nom, categorie, prix, quantite = parts
                stock.append(Produit(nom, categorie, float(prix), int(quantite)))


# Sauvegarder les données dans un fichier
def sauvegarder_stock():
    with open(FICHIER, "w") as f:
        for produit in stock:
            f.write(f"{produit.nom},{produit.categorie},{produit.prix},{produit.quantite}\n")
    print("Stock sauvegardé avec succès.")


# Ajouter un produit
def ajouter_produit():
    nom = input("Entrez le nom du produit : ")
    categorie = input("Entrez la catégorie : ")
    prix = float(input("Entrez le prix : "))
    quantite = int(input("Entrez la quantité : "))
    stock.append(Produit(nom, categorie, prix, quantite))
    print("Produit ajouté avec succès !")


# Afficher le stock
def afficher_stock():
    if not stock:
        print("Le stock est vide.")
        return

    print("\n--- État actuel du stock ---")
    for i, produit in enumerate(stock, 1):
        print(f"{i}. {produit}")


# Rechercher un produit par son nom
def rechercher_produit():
    nom = input("Entrez le nom du produit à rechercher : ")
    for produit in stock:
        if produit.nom.lower() == nom.lower():
            print("Produit trouvé :")
            print(produit)
            return
    print("Produit non trouvé.")


# Modifier un produit
def modifier_produit():
    nom = input("Entrez le nom du produit à modifier : ")
    for produit in stock:
        if produit.nom.lower() == nom.lower():
            produit.nom = input("Entrez le nouveau nom : ")
            produit.categorie = input("Entrez la nouvelle catégorie : ")
            produit.prix = float(input("Entrez le nouveau prix : "))
            produit.quantite = int(input("Entrez la nouvelle quantité : "))
            print("Produit modifié avec succès.")
            return
    print("Produit non trouvé.")


# Supprimer un produit
def supprimer_produit():
    nom = input("Entrez le nom du produit à supprimer : ")
    for i, produit in enumerate(stock):
        if produit.nom.lower() == nom.lower():
            del stock[i]
            print("Produit supprimé avec succès.")
            return
    print("Produit non trouvé.")


# Mettre à jour la quantité d'un produit
def mettre_a_jour_quantite():
    nom = input("Entrez le nom du produit : ")
    for produit in stock:
        if produit.nom.lower() == nom.lower():
            produit.quantite = int(input("Entrez la nouvelle quantité : "))
            print("Quantité mise à jour avec succès.")
            return
    print("Produit non trouvé.")


# Fonction principale
def main():
    charger_stock()
    while True:
        afficher_menu()
        choix = input("Entrez votre choix : ")
        if choix == "1":
            ajouter_produit()
        elif choix == "2":
            afficher_stock()
        elif choix == "3":
            sauvegarder_stock()
        elif choix == "4":
            modifier_produit()
        elif choix == "5":
            supprimer_produit()
        elif choix == "6":
            rechercher_produit()
        elif choix == "7":
            mettre_a_jour_quantite()
        elif choix == "8":
            print("Merci d'avoir utilisé le programme. Au revoir !")
            sauvegarder_stock()
            break
        else:
            print("Choix invalide, veuillez réessayer.")


if __name__ == "__main__":
    main()
